qdev_wrappers/fitting/Converter.py

```diff
@@ -0,0 +1,236 @@
+import qcodes as qc
+import numpy as np
+from os.path import sep
+from qcodes.dataset.data_export import get_data_by_id
+from qcodes.dataset.sqlite_base import connect
+from qdev_wrappers.file_setup import CURRENT_EXPERIMENT
+from qdev_wrappers.show_num import check_experiment_is_initialized
+
+
+def data_to_dict(id, samplefolder=None, datatype='SQL'):
+    if datatype == 'qcodes_legacy':
+        converter = Legacy_Converter()
+
+    if datatype == 'SQL':
+        converter = SQL_Converter()
+
+    return converter.convert(id, samplefolder)
+
+
+class DataConverter():
+    
+    def convert(self, id, samplefolder = None):
+        
+        data = self.find_data(id, samplefolder) 
+        data_dict = self.make_data_dictionary(data)
+
+        exp_id = self.find_experiment(id)
+        dependencies = self.find_dependencies(id, data)  #SQL needs only id number to find dependencies, Legacy needs only data
+        all_variables = self.find_variables(data)
+
+        data_dict['exp_id'] = exp_id
+        data_dict['run_id'] = id
+        data_dict['dependencies'] = dependencies
+        data_dict['variables'] = all_variables
+        
+        return data_dict
+    
+
+
+class SQL_Converter(DataConverter):
+    
+    def find_data(self, id, samplefolder):  
+        
+            '''sample folder does nothing except act as placeholder.                    
+            This function currently proceeds under the assumption that the experiment is set up, and doesn't
+            check anything to confirm that. Also, I think this doesn't return any error if it can't find 
+            the dataset with that id number.'''
+          
+            data = get_data_by_id(id)                   
```

Comment:
does this assume that the database is stored directly in the samplefolder? I can test this if you don't get a chance :)

Reply:
get_data_by_id uses qc.config to get the database location :)